docs/notebooks/recovering_from_errors.pct.py

```diff
@@ -124,6 +124,24 @@ def __call__(self, x):
     )
     plot_bo_points(data.query_points.numpy(), ax[0, 0], 5, arg_min_idx)
 
+# %% [markdown]
+# ## Out of memory errors
+#
+# At the moment, trieste's `OptimizationResult` doesn't allow you to recover from Out Of Memory errors, which normally result in the Python process shutting down. One possible cause of memory errors is performing a gradient-based optimization over a large dataset. To work around this, you can specify that evaluations of the acquisition function be batchified: this splits them (on the first dimension) into batches of a given size, then stitches them back together. To do this, you need to provide an explicit batchified optimizer and specify a desired batch size.
+
```

Comment:
One possible cause of memory errors is trying to evaluate an acquisition function over a large dataset, e.g. when initializing our gradient-based optimizers.